benchmark/single-source/BufferFill.swift

```diff
@@ -0,0 +1,121 @@
+//===--- BufferFill.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let BufferFill = [
+  BenchmarkInfo(name: "BufferFillFromSlice",
+                runFunction: BufferFillFromSliceExecute,
+                tags: [.validation, .api],
+                setUpFunction: BufferFillFromSliceSetup,
+                tearDownFunction: BufferFillFromSliceTeardown),
+  BenchmarkInfo(name: "RawBufferCopyBytes",
+                runFunction: RawBufferCopyBytesExecute,
+                tags: [.validation, .api],
+                setUpFunction: RawBufferCopyBytesSetup,
+                tearDownFunction: RawBufferCopyBytesTeardown),
+  BenchmarkInfo(name: "RawBufferInitializeMemory",
+                runFunction: RawBufferInitializeMemoryExecute,
+                tags: [.validation, .api],
+                setUpFunction: RawBufferInitializeMemorySetup,
+                tearDownFunction: RawBufferInitializeMemoryTeardown),
+]
+
+let c = 100_000
+let a = Array(0..<c)
+var b: UnsafeMutableBufferPointer<Int> = .init(start: nil, count: 0)
+
+public func BufferFillFromSliceSetup() {
+  assert(b.baseAddress == nil)
+  b = .allocate(capacity: c)
+}
+
+public func BufferFillFromSliceTeardown() {
+  b.deallocate()
+  b = .init(start: nil, count: 0)
+}
+
+@inline(never)
+public func BufferFillFromSliceExecute(N: Int) {
+  // Measure performance when filling an UnsafeBuffer from a Slice
+  // of a Collection that supports `withContiguousStorageIfAvailable`
+  // See: https://bugs.swift.org/browse/SR-14491
+
+  for _ in 0..<N {
+    let slice = Slice(base: a, bounds: a.indices)
+    var (iterator, copied) = b.initialize(from: slice)
+    blackHole(b)
+    guard copied == a.count, iterator.next() == nil else { break }
```

Comment:
Why not just call `CheckResults` here? If this early exit is ever taken, it would lead to invalid results.
```suggestion
    CheckResults(copied == a.count && iterator.next() == nil)
```